src/main/java/mServer/crawler/AddToFilmlist.java

```diff
@@ -5,176 +5,270 @@
+        Log.sysLog("In Liste einsortiert: " + treffer);
         Log.sysLog("");
-        return treffer.get();
+        return treffer;
     }
 
-    private boolean stopOld = false;
+    /**
+     * 
+     * Add all local thread results to the filmlist.
+     *
+     * @return the total number of entries found.
+     */
+    private int retrieveThreadResults() {
+        int treffer = 0;
+        for (ImportOldFilmlistThread t : threadList) {
+            final ArrayList<DatenFilm> localList = t.getLocalAddList();
+            if (MserverDaten.debug)
+                Log.sysLog("Thread " + t.getName() + " list size: " + localList.size());
+            vonListe.addAll(localList);
+            localList.clear();
+            treffer += t.getTreffer();
+        }
+        return treffer;
+    }
 
-    private class AddOld implements Runnable {
+    private class ImportOldFilmlistThread extends Thread {
 
-        private DatenFilm film;
         private final ListeFilme listeOld;
-        private final int MIN_SIZE_ADD_OLD = 5; //REST eh nur Trailer
+        private final ArrayList<DatenFilm> localAddList = new ArrayList<>((vonListe.size() / NUMBER_OF_THREADS) + 500);
+        private int treffer = 0;
+        private OkHttpClient client = null;
 
-        public AddOld(ListeFilme listeOld) {
+        public ImportOldFilmlistThread(ListeFilme listeOld, OkHttpClient client) {
             this.listeOld = listeOld;
             threadCounter.incrementAndGet();
+            this.client = client;
+        }
+
+        public int getTreffer() {
+            return treffer;
+        }
+
+        public ArrayList<DatenFilm> getLocalAddList() {
+            return localAddList;
+        }
+
+        private void addOld(DatenFilm film) {
+            treffer++;
+            film.init();
+
+            localAddList.add(film);
+        }
+
+        private synchronized DatenFilm popOld(ListeFilme listeOld) {
+            if (!listeOld.isEmpty()) {
+                return listeOld.remove(0);
+            } else
+                return null;
         }
 
         @Override
         public void run() {
-            EtmPoint performancePoint = etmMonitor.createPoint("AddOld:run");
-            while (!stopOld && (film = popOld(listeOld)) != null) {
-                long size = FileSize.laengeLong(film.arr[DatenFilm.FILM_URL]);
-                if (size > MIN_SIZE_ADD_OLD) {
-                    addOld(film);
+
+            DatenFilm film;
+            while (!isInterrupted() && (film = popOld(listeOld)) != null) {
+                final String url = film.arr[DatenFilm.FILM_URL];
+                if (film.arr[DatenFilm.FILM_GROESSE].isEmpty()) {
+                    Request request = new Request.Builder().url(url).head().build();
+                    try (Response response = client.newCall(request).execute();
+                         ResponseBody body = response.body()) {
+                        if (response.isSuccessful()) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS134)